src/lib/schematics/ng-update/upgrade-rules/package-imports-v8/updateAngularMaterialImportsRule.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * This schematic corrects to importing symbols from Angular Material through
+ * subpackages (e.g. @angular/material/button) rather than the top level
+ * package (e.g. @angular/material).
+ */
+import * as Lint from 'tslint';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+
+/** The import path for Angular Material top level module. */
+const ANGULAR_MATERIAL_IMPORT_PATH = `'@angular/material'`;
```

Comment:
Optional: Use this from `src/lib/schematics/ng-update/typescript/module-specifiers.ts`? (so that is consistent). You can construct the Regex from that constant or just move the regex to that `module-specifiers.ts` file if you want.